core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/stack/upscale/StackUpscaleService.java

```diff
@@ -206,21 +206,12 @@ private void validateResourceResults(StackScalingFlowContext context, Exception
 
     private Long getFirstValidPrivateId(List<InstanceGroup> instanceGroups) {
         LOGGER.debug("Get first valid PrivateId of instanceGroups");
-        long highest = 0;
-        for (InstanceGroup instanceGroup : instanceGroups) {
-            LOGGER.debug("Checking of instanceGroup: {}", instanceGroup.getGroupName());
-            for (InstanceMetaData metaData : instanceGroup.getAllInstanceMetaData()) {
-                Long privateId = metaData.getPrivateId();
-                LOGGER.debug("InstanceMetaData metaData: privateId: {}, instanceGroupName: {}, instanceId: {}, status: {}",
-                        privateId, metaData.getInstanceGroupName(), metaData.getInstanceId(), metaData.getInstanceStatus());
-                if (privateId == null) {
-                    continue;
-                }
-                if (privateId > highest) {
-                    highest = privateId;
-                }
-            }
-        }
+        long highest = instanceGroups.stream()
```

Comment:
so this will produce invalid output. if we have highest ID `0` then we will return `0` as first valid id. this should be an `Optional<Long>` and the return look like:
```
return highest.isPresent() ? highest.get() + 1 : 0;
```
also we might have a test for this method to be sure we don't break it in the future